pypdf/_writer.py

```diff
@@ -2421,17 +2421,34 @@ def pageMode(self, mode: PagemodeType) -> None:  # deprecated
         deprecation_with_replacement("pageMode", "page_mode", "3.0.0")
         self.page_mode = mode
 
-    def add_annotation(self, page_number: int, annotation: Dict[str, Any]) -> None:
+    def add_annotation(
+        self,
+        page_number: Union[int, PageObject],
+        annotation: Dict[str, Any],
+    ) -> DictionaryObject:
+        """
+        Add a single annotation to the page. Must be a new annotation (can not be recycled)
+
+        Args:
+            page: page object or number (used to be pagenumber : deprecated)
```

Comment:
I haven't seen any version where it was `pagenumber`. I've introduced this via https://github.com/py-pdf/pypdf/pull/1120/files#diff-5223fb0f4252076a23d2cc060dc32a4b58acb3b97c57978cd8afb8da16407e46

```suggestion
            page_number: page object or number
```